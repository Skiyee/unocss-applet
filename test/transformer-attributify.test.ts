import { promises as fs } from 'fs'
import path from 'path'
import { describe, expect, test } from 'vitest'
import type { SourceCodeTransformer } from '@unocss/core'
import { createGenerator } from '@unocss/core'
import MagicString from 'magic-string'
import transformerAttributify from '@unocss-applet/transformer-attributify'
import presetUno from '@unocss/preset-uno'

describe('transformer-attributify', async () => {
  const content = await fs.readFile(
    path.resolve(__dirname, './fixtures/attributify.vue'),
  )

  async function transform(code: string, transformer: SourceCodeTransformer) {
    const s = new MagicString(code)

    const uno = createGenerator({
      presets: [
        presetUno(),
      ],
      transformers: [
        transformer,
      ],
    })
    await transformer.transform(s, 'foo.js', { uno, tokens: new Set() } as any)
    const result = s.toString()
    const { css } = await uno.generate(result, { preflights: false })
    return {
      code: result,
      css,
    }
  }

  test('basic', async () => {
    const transformer = transformerAttributify()
    const result = await transform(content.toString(), transformer)
    expect(result).toMatchSnapshot()
  })

  test('prefixedOnly', async () => {
    const transformer = transformerAttributify({ prefixedOnly: true })
    const result = await transform(content.toString(), transformer)
    expect(result).toMatchSnapshot()
  })
})
