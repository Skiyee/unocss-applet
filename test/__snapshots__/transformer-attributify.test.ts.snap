// Vitest Snapshot v1

exports[`transformer-attributify > basic 1`] = `
{
  "code": "<script setup lang=\\"ts\\">
import { ref } from 'vue'
const bool = ref<boolean>()
</script>

<template>
  <div class=\\"text-center aaa p-4 text-blue\\">
    <div  class=\\"text-4xl text-green p-2 ma\\"/>
    <div :class=\\"\`p-2.5 \${bool ? 'p-0.5' : ''}\`\\" :hover-class=\\"['!bg-green']\\"  class=\\"m-2\\"/>
    <div class=\\"p-1 flex flex-col flex-gap-1 items-center\\" />
    <div class=\\"b b-green text-right h-10 !text-pink\\">
      123456789
    </div>
    <div class=\\"text-purple b b-green\\">
      123456789
    </div>
  </div>
</template>
",
  "css": "/* layer: default */
.m-2{margin:0.5rem;}
.ma{margin:auto;}
.h-10{height:2.5rem;}
.flex{display:flex;}
.flex-col{flex-direction:column;}
.items-center{align-items:center;}
.flex-gap-1{grid-gap:0.25rem;gap:0.25rem;}
.b{border-width:1px;}
.b-green{--un-border-opacity:1;border-color:rgba(74,222,128,var(--un-border-opacity));}
.\\\\!bg-green{--un-bg-opacity:1 !important;background-color:rgba(74,222,128,var(--un-bg-opacity)) !important;}
.p-0\\\\.5{padding:0.125rem;}
.p-1{padding:0.25rem;}
.p-2{padding:0.5rem;}
.p-2\\\\.5{padding:0.625rem;}
.p-4{padding:1rem;}
.text-center{text-align:center;}
.text-right{text-align:right;}
.text-4xl{font-size:2.25rem;line-height:2.5rem;}
.\\\\!text-pink{--un-text-opacity:1 !important;color:rgba(244,114,182,var(--un-text-opacity)) !important;}
.text-blue{--un-text-opacity:1;color:rgba(96,165,250,var(--un-text-opacity));}
.text-green{--un-text-opacity:1;color:rgba(74,222,128,var(--un-text-opacity));}
.text-purple{--un-text-opacity:1;color:rgba(192,132,252,var(--un-text-opacity));}",
}
`;

exports[`transformer-attributify > prefixedOnly 1`] = `
{
  "code": "<script setup lang=\\"ts\\">
import { ref } from 'vue'
const bool = ref<boolean>()
</script>

<template>
  <div class=\\"text-center aaa\\" p=\\"4\\" text-blue>
    <div text=\\"4xl\\" text-green ma  class=\\"p-2\\"/>
    <div :class=\\"\`p-2.5 \${bool ? 'p-0.5' : ''}\`\\" m-2 :hover-class=\\"['!bg-green']\\" />
    <div flex=\\"~ col gap-1\\" class=\\"p-1\\" items-center />
    <div text-right h-10 text=\\"!pink\\" class=\\"b b-green\\">
      123456789
    </div>
    <div class=\\"text-purple b b-green\\">
      123456789
    </div>
  </div>
</template>
",
  "css": "/* layer: default */
.m-2{margin:0.5rem;}
.ma{margin:auto;}
.h-10{height:2.5rem;}
.items-center{align-items:center;}
.gap-1{grid-gap:0.25rem;gap:0.25rem;}
.b{border-width:1px;}
.b-green{--un-border-opacity:1;border-color:rgba(74,222,128,var(--un-border-opacity));}
.\\\\!bg-green{--un-bg-opacity:1 !important;background-color:rgba(74,222,128,var(--un-bg-opacity)) !important;}
.p-0\\\\.5{padding:0.125rem;}
.p-1{padding:0.25rem;}
.p-2{padding:0.5rem;}
.p-2\\\\.5{padding:0.625rem;}
.text-center{text-align:center;}
.text-right{text-align:right;}
.text-green{--un-text-opacity:1;color:rgba(74,222,128,var(--un-text-opacity));}
.text-purple{--un-text-opacity:1;color:rgba(192,132,252,var(--un-text-opacity));}",
}
`;
